src/utils/contract-functions.ts

```diff
@@ -42,40 +42,158 @@ export const getRacesWithPagination = async(userAddr: `0x${string}`, from: numbe
     return data;
 }
 
+// get race by id
+export const getRaceById = async(raceId: number) => {
+    const data = await readContract(config, {
+        address: BLOCK_SHEEP_CONTRACT,
+        abi: BlockSheepAbi,
+        functionName: "getRaces",
+        args: [BigInt(raceId)]
+    });
+
+    
+    const questionsData = await readContracts(config, {
+        // @ts-ignore
+        contracts: data[5].map(gameId => {
+            return {
+                address: BLOCK_SHEEP_CONTRACT,
+                abi: BlockSheepAbi,
+                functionName: "getQuestions",
+                args: [BigInt(raceId), BigInt(gameId)]
+            }
+        }) 
+    });
+
+    return {
+        race: data,
+        //@ts-ignore
+        questionsByGames: questionsData.map(i => i.result)
+    };
+}
+
 // enter the race
 export const registerOnTheRace = async(raceId: number, walletAddress: `0x${string}`) => {
-    const data = await writeContract(config, {
-        account: walletAddress,
+    const COST = await retreiveCOST();
+    console.log(BigInt(Number(COST) * 3));
```

Comment:
Assuming that "3" is a numOfQuetions, it would be better to fetch this data from smart-contract